src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationTransformationTest.kt

```diff
@@ -102,6 +102,64 @@ class AdjudicationTransformationTest {
     }
   }
 
+  @Test
+  fun `will use the withOther offence code if this adjudication is the other prisoner in an incident`() {
+    val dpsAdjudication = dpsAdjudication().copy(
+      reportedAdjudication = dpsAdjudication().reportedAdjudication.copy(
+        incidentRole = IncidentRoleDto(
+          roleCode = "25c",
+          offenceRule = OffenceRuleDetailsDto(
+            paragraphNumber = "25(c)",
+            paragraphDescription = "Assists another prisoner to commit, or to attempt to commit, any of the foregoing offences:",
+          ),
+          associatedPrisonersName = "A4323ZA",
+        ),
+        offenceDetails = OffenceDto(
+          offenceCode = 1002,
+          offenceRule = OffenceRuleDto(
+            paragraphNumber = "1",
+            paragraphDescription = "Commits any assault",
+            nomisCode = "51:1B",
+            withOthersNomisCode = "51:25D",
+          ),
+          victimPrisonersNumber = "A1234AA",
+        ),
+      ),
+    )
+    val nomisAdjudication = dpsAdjudication.toNomisAdjudication()
+
+    assertThat(nomisAdjudication.charges).hasSize(1)
+    assertThat(nomisAdjudication.charges[0].offenceCode).isEqualTo("51:25D")
+    assertThat(nomisAdjudication.charges[0].offenceId).isEqualTo("2234567/1")
+  }
+
+  @Test
+  fun `assume there is a withOther offence code if this adjudication is the other prisoner in an incident`() {
```

Comment:
I was wondering about the !!s above... why would it be nullable in the API response but never in real life?

Reply:
It is the Adjudication DTO that can be nullable (this is generated from swagger )  when there is a mapping issue why creating an adjudication, but by the time it is created and we get event it should be OK